### DISCUSSION TITLE
PR merge policy

### DISCUSSION THREAD

**Discussion by whereswaldon** [go-text/typesetting#36] [General]
The `go-text` project is governed by a trio of interested folks known as the "governing council". This council is currently made up of:

- @benoitkugler 
- @andydotxyz 
- @whereswaldon 

We usually made policy decisions with majority concensus, but I'm slightly unclear on how many of us need to approve a PR in order to merge it. Is it also a simple majority? I just wanted to broach the question so we can decide and maybe implement the results as a branch protection rule.

I propose the following, but am happy to discuss alternatives:

PRs may be merged if all of the following are true:

- They do not have changes requested by a council member
- They are approved by at least two council members who are not authors of the PR

What do you think?

**Comment by andydotxyz** [comment]
I agree this should be formalised, but if the above rules are followed won't it be impossible for a PR which one of us wrote to be accepted?

Could maybe be simplified to 2 "council" :+1: required for any PR?

**Reply by whereswaldon** [reply]
I think perhaps you're reading them differently than I meant. So if I (for example) open a PR, the rules state that both you and Benoit need to approve.

Another example: If a random community member opens a PR, it needs two of us to approve *and* the third member to *not* request changes.

I guess it can be rephrased as: No PR gets merged if one of us is actively opposed to its current state. If two approve and the other is neutral, it can merge.

**Comment by andydotxyz** [comment]
Apologies your are quite right - I read "changes request" as "changes contributed" - been a long day.
Sounds good then.

**Comment by benoitkugler** [comment]
Sounds good to me, it's in line with what we have done so far.

**Comment by whereswaldon** [comment]
Okay, I've configured a branch protection rule for `main` requiring two approvals to merge into it and preventing changes from bypassing PRs. Take a peek at the settings and let me know if you have any concerns.

**Reply by andydotxyz** [reply]
I think those settings look like any member of the project can approve, rather than just the council members you listed above

**Reply by whereswaldon** [reply]
Yeah, that's true. What's the best way to enforce that it needs to be council members? The code owners feature?

**Reply by andydotxyz** [reply]
I don't know of any clever way to do it.
The only way I can think is that the accounts you don't want to be included don't get "write" permissions on the repos in question.
Not sure if that is a big deal or not, and there may be a better way.

**Reply by whereswaldon** [reply]
I've opened #38 to get a little closer to enforced policy. I know we can require approval from owners specifically, but I'm not sure if we can require approval from two owners. Either way, it's an improvement.

**Reply by whereswaldon** [reply]
Okay, a Code Owner approval is now required.

**Comment by mengtianyoo** [comment]
我已成功收到您的邮件
